tests/test_flow/test_api.py

```diff
@@ -376,6 +381,36 @@ def test_get_formats(preset_flow):
         assert list(p_series_index_df['q']) == [5]
 
 
+def test_get_modes_exclude_object(preset_flow):
+    # TODO: should we raise a warning for ValueProvider entities when
+    # queried for a mode that implies the existence of a persisted file?
+    flow = preset_flow
+
+    for mode in [Path, 'path']:
+        y_path = flow.get('y', mode=mode)
+        assert y_path is None
+
+        y_fxn_path = flow.get('y_fxn', mode=mode)
+        assert isinstance(y_fxn_path, Path)
+        assert y_fxn_path.is_file()
+
+    y_filecopier = flow.get('y', mode='FileCopier')
+    assert isinstance(y_filecopier, bn.util.FileCopier)
+    assert y_filecopier.src_file_path is None
+
+    y_fxn_filecopier = flow.get('y_fxn', mode='FileCopier')
+    assert isinstance(y_fxn_filecopier, bn.util.FileCopier)
+    assert y_fxn_filecopier.src_file_path.is_file()
+
+    y_filename = flow.get('y', mode='filename')
+    assert y_filename == 'None'
+    # TODO: maybe don't stringify None if there is no filename to return?
```

Comment:
Yeah, returning a string here doesn't seem like what I'd expect as a user.  We could return `None` itself but I think an exception would make more sense.